src/components/PeopleTable/PeopleTable.tsx

```diff
@@ -0,0 +1,176 @@
+import { useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { Person } from '../../types';
+import { NoPeopleMessage } from '../NoPeopleMessage/NoPeopleMessage';
+import { PersonLink } from '../PersonLink/PersonLink';
+import { Sex } from '../../types/Sex';
+import { SearchLink } from '../SearchLink';
+import { SortCategory } from '../../types/sortCategory';
+import { handleSort } from '../../helpers/handleSort';
+import { getSortIconClass } from '../../helpers/getSortIconClass';
+
+type Props = {
+  people: Person[];
+};
+
+export const PeopleTable: React.FC<Props> = ({ people }) => {
+  const [filteredPeople, setFilteredPeople] = useState<Person[]>(people);
+
+  const [searchParams] = useSearchParams();
+
+  useEffect(() => {
+    let filteredList = [...people];
+    const query = searchParams.get('query') || '';
+    const centuries = searchParams.getAll('centuries') || [];
+
+    const lowerCaseQuery = query.toLowerCase().trim();
+
+    filteredList = people.filter(
+      (person) => person.name.toLowerCase().includes(lowerCaseQuery)
+        || person.motherName?.toLowerCase().includes(lowerCaseQuery)
+        || person.fatherName?.toLowerCase().includes(lowerCaseQuery),
+    );
+
+    if (!query) {
+      searchParams.delete('query');
+    }
+
+    if (centuries.length) {
+      filteredList = filteredList.filter(
+        person => centuries.includes(String(Math.ceil(+person.born / 100))),
+      );
+    }
+
+    switch (searchParams.get('sex')) {
+      case Sex.MALE: {
+        filteredList = filteredList.filter((person) => person.sex === Sex.MALE);
+        break;
+      }
+
+      case Sex.FEMALE: {
+        filteredList = filteredList.filter(
+          (person) => person.sex === Sex.FEMALE,
+        );
+        break;
+      }
+
+      default: {
+        break;
+      }
+    }
+
+    const sortBy = searchParams.get('sort');
+    const sortOrder = searchParams.get('order');
+
+    switch (sortBy) {
+      case SortCategory.name: {
+        filteredList.sort((a, b) => a.name.localeCompare(b.name));
+        break;
+      }
+
+      case SortCategory.born: {
```

Comment:
case SortCategory.born: {
        filteredList.sort((a, b) => a.born - b.born);
        break;
      }

      case SortCategory.died: {
        filteredList.sort((a, b) => a.died - b.died);
        break;
      }

Don't repeat yourself. You can wtite it like "people.sort((a, b) => order * (a[sortParam] - b[sortParam]))"

Reply:
right